### PR TITLE
MONet Bundle Integration into MONAI Deploy

### PR DESCRIPTION
**Pull request by SimoneBendazzoli93**
This PR introduces support for the MONet Bundle (an nnUNet wrapper for the MONAI Bundle) into MONAI Deploy.

Key Features:

- Added a new operator: **`MONetBundleInferenceOperator`**, extending **`MonaiBundleInferenceOperator`**

- Included an example application demonstrating spleen segmentation using the **`MONetBundleInferenceOperator`**

## Summary by CodeRabbit

* **New Features**
  * Added MONetBundleInferenceOperator — a specialized inference operator with nnUNet-style model support, multimodal input handling, automatic predictor setup, and streamlined prediction behavior.
* **Bug Fixes**
  * Fixed YAML extension handling when reading bundle configurations.
  * Normalized missing input metadata to ensure consistent validation and processing.
* **Chores**
  * Exported MONetBundleInferenceOperator in the package public API.

### CODE REVIEW

File: devel/monet_bundle_inference_operator.py
Comment (chezhia) [inline devel/monet_bundle_inference_operator.py:1]: This breaks the existing project structure by introducing a new folder 'devel', can this not be part of the file under ''operators" folder?  - Also, can you add some links in the docstrings on how to generate the MONET bundle. Does this currently support all versions of nnunet?

**Review by chezhia — CHANGES_REQUESTED** [review]

File: devel/monet_bundle_inference_operator.py
Comment (SimoneBendazzoli93) [reply devel/monet_bundle_inference_operator.py:1]: The devel folder was accidentally included in the PR. I have now removed, adding also some references to the MONet Bundle in the docstrings

File: monai/deploy/operators/monet_bundle_inference_operator.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+        self._nnunet_predictor: torch.nn.Module = None
+
+    def _init_config(self, config_names):
+
+        super()._init_config(config_names)
+        parser = get_bundle_config(str(self._bundle_path), config_names)
+        self._parser = parser
+
+        self._nnunet_predictor = parser.get_parsed_content("network_def")
```
[the remark this answers is not in the document]
Comment (chezhia) [reply monai/deploy/operators/monet_bundle_inference_operator.py:63]: This suggestion from copilot needs review - If the parent class is initializing the model properly, we can accept the suggestion, but I am not sure if get_bundle_config is needed to apply any config patch.

[the remark this answers is not in the document]
Comment (chezhia) [reply monai/deploy/operators/monet_bundle_inference_operator.py:63]: @SimoneBendazzoli93  - please review this suggestion above

**Comment by chezhia** [comment]
@SimoneBendazzoli93  It looks like the DCO (Developer Certificate of Origin) check is failing. To fix this, please ensure all your commits are signed off.

You can do this by amending your previous commits using:
git commit --amend --signoff

Or, if you have multiple commits, you can perform an interactive rebase:
git rebase -i main --signoff

Then, force-push the changes to the branch. This is required for the PR to be mergedr

File: monai/deploy/operators/monet_bundle_inference_operator.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        it concatenates the data with the main input data."""
+
+        self._set_model_network(self._model_network)
+
+        if len(kwargs) > 0:
+            multimodal_data = {"image": data}
+            for key in kwargs.keys():
+                if isinstance(kwargs[key], MetaTensor):
+                    multimodal_data[key] = ResampleToMatch(mode="bilinear")(kwargs[key], img_dst=data)
+            data = ConcatItemsd(keys=list(multimodal_data.keys()), name="image")(multimodal_data)["image"]
```
[the remark this answers is not in the document]
Comment (chezhia) [reply monai/deploy/operators/monet_bundle_inference_operator.py:94]: I think this can be ignored

**Review by chezhia — CHANGES_REQUESTED** [review]
A couple of items identified by copilot needs review from author. Accepted a few minor suggestions.
